zenodo/modules/records/views.py

```diff
@@ -423,3 +426,37 @@ def community_curation(record, user):
 def record_communities():
     """Context processor for community curation for given record."""
     return dict(community_curation=community_curation)
+
+
+def record_thumbnail(pid, record, **kwargs):
+    """Provide easy access to the thumbnail of a record for predefined sizes.
+
+    We consider the thumbnail of the record as the first image in the files
+    iterator.
+    """
+    if not has_record_perm(current_user, record, 'read-files'):
+        abort(400, 'Bad Request')
```

Comment:
Check what the return code is for getting a closed-access file for an anonymous user, and replicate this here